### PR TITLE
feat: sdk 1954 misc issues remove external ids and collect ad

### PR DESCRIPTION
**Pull request by itsdebs**
**Fixes** # (*issue*)
Remove external ids from all messages except identify.
Collection of advertising Id is fixed. Previously the default executor for ad id collection was null.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] Version upgraded (project, README, gradle, podspec etc)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by 1abhishekpandey — CHANGES_REQUESTED** [review]
I tested this and there is a bug now `externalIds` value set using `identify` event is not being persisted. Please fix this!

File: android/src/main/java/com/rudderstack/android/internal/plugins/PlatformInputsPlugin.kt [hunk 1]
```diff
@@ -108,8 +108,24 @@ internal class PlatformInputsPlugin : Plugin, LifecycleListenerPlugin {
...
+        if(!isOnClassPath("com.google.android.gms.ads.identifier.AdvertisingIdClient")){
+            _analytics?.currentConfiguration?.logger?.debug(log = "Not collecting advertising ID because "
+                    + "com.google.android.gms.ads.identifier.AdvertisingIdClient "
+                    + "was not found on the classpath."
+            )
+            return
+        }
         _analytics?.currentConfigurationAndroid?.advertisingIdFetchExecutor?.submit {
-            val adId = getGooglePlayServicesAdvertisingID() ?: getAmazonFireAdvertisingID()
+            val adId = try {
```
Comment (ChryssaAliferi) [inline android/src/main/java/com/rudderstack/android/internal/plugins/PlatformInputsPlugin.kt:119]: 👍

File: samples/sample-kotlin-android/build.gradle.kts [hunk 2]
```diff
@@ -124,6 +124,8 @@ dependencies {
     implementation("androidx.compose.runtime:runtime-livedata:1.4.3")
     implementation("androidx.compose.ui:ui-tooling-preview:1.4.3")
     implementation("androidx.lifecycle:lifecycle-runtime-ktx:2.6.2")
+    // adding play services to generate advertising id
+    implementation("com.google.android.gms:play-services-ads:22.1.0")
```
Comment (ChryssaAliferi) [inline samples/sample-kotlin-android/build.gradle.kts:128]: 👍  Great addition to the sample application!! 

But why are we adding an old version of `com.google.android.gms:play-services-ads`?
The latest version is: 
`com.google.android.gms:play-services-ads:23.1.0`

**Review by ChryssaAliferi — APPROVED** [review]

File: android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt [hunk 3]
```diff
@@ -14,97 +14,183 @@
...
+        `when`(chain.message()).thenReturn(aliasMessage)
+        plugin.intercept(chain)
+        val messageCaptor = argumentCaptor<AliasMessage>()
+
+        verify(chain).proceed(messageCaptor.capture())
+        val capturedMessage = messageCaptor.lastValue
+        assertEquals("newUserId", capturedMessage.userId)
+    }
+
+ /*   @Test
```
Comment (1abhishekpandey) [inline android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt:110]: If all the commented-out tests are not needed, can you please consider removing them?

File: android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt [hunk 4]
```diff
@@ -14,97 +14,183 @@
...
-//            storage.externalIds!![0], allOf(aMapWithSize(1),
-//            hasEntry("dest-1", "dest-1-id")))
+    fun `intercept should proceed with modified message for IdentifyMessage with anonymousId`() {
+        val identifyMessage = IdentifyMessage.create(traits = mapOf("anonymousId" to null, "userId" to "userId"), timestamp = RudderUtils.timeStamp)
+        `when`(chain.message()).thenReturn(identifyMessage)
+
+        plugin.intercept(chain)
+        val messageCaptor = argumentCaptor<IdentifyMessage>()
+        verify(chain).proceed(messageCaptor.capture())
+        println("118,${messageCaptor.allValues}")
```
Comment (1abhishekpandey) [inline android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt:93]: Please remove this `println` statement.

**Comment by 1abhishekpandey** [comment]
Regarding advertisingId changes, as half of the changes is in the migration branch (https://github.com/rudderlabs/rudder-sdk-android/pull/437). I'll test this feature once all the code related to advertisingId gets merged into `feat/android`.

File: models/src/main/java/com/rudderstack/models/Extensions.kt [hunk 5]
```diff
@@ -32,3 +32,68 @@ val MessageContext.externalIds: List<Map<String, String>>?
...
+ * @param item The item to be checked
+ * @return true if the map with same keys are present, false otherwise
+ */
+infix fun <K, V> Iterable<Map<K, V>>.inWrtKeys(item: Map<K, V>): Boolean {
+    this.toSet().forEach {
+        if (it.keys.containsAll(item.keys))
+            return true
+    }
+    return false
+}
```
[the remark this answers is not in the document]
Comment (itsdebs) [reply models/src/main/java/com/rudderstack/models/Extensions.kt:99]: This is an util function, let's keep it for now. If this doesn't find a use case till last iteration, will remove it.

File: android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt (hunk 3 above)
Comment (itsdebs) [reply android/src/test/java/com/rudderstack/android/plugins/ExtractStatePluginTest.kt:110]: No, we need to fix them later for better test coverage. Will get back to this when other PRs are merged.

**Review by 1abhishekpandey — APPROVED** [review]

**Pull request merged**